### PR TITLE
Move well-known env vars and config map keys into constants

### PR DESCRIPTION
**Pull request by nicmorales9**
# Description

fixes https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1412 and https://github.com/FoundationDB/fdb-kubernetes-operator/issues/2029

I did my best to add meaningful comment descriptions, almost entirely lifted from descriptions elsewhere in the operator or in the FoundationDB repo itself - would be great to check those over

## Type of change

*Please select one of the options below.*

- Other (code tidying)

## Discussion

*Are there any design details that you would like to discuss further?*
No

## Testing

*Please describe the tests that you ran to verify your changes. Unit tests?
Manual testing?*
I ran the unit tests

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*
No

## Documentation

N/A

## Follow-up

N/A

### CODE REVIEW

File: api/v1beta1/foundationdbcluster_types_test.go [hunk 1]
```diff
@@ -2411,11 +2413,11 @@ var _ = Describe("[api] FoundationDBCluster", func() {
 			Entry("With a placeholder",
 				testCase{
 					processAddr: ProcessAddress{
-						StringAddress: "$POD_IP",
+						StringAddress: fmt.Sprintf("$%s", fdbv1beta2.EnvNamePodIP),
```
Comment (johscheuer) [inline api/v1beta1/foundationdbcluster_types_test.go:2416]:
```suggestion
						StringAddress: string(fdbv1beta2.EnvNamePodIP),
```

Is there a reason to use `fmt.Sprintf`?

File: api/v1beta2/foundationdb_constants.go [hunk 2]
```diff
@@ -32,4 +32,20 @@ const (
 
 	// NoneFaultDomainKey represents the none fault domain, where every Pod is a fault domain.
 	NoneFaultDomainKey = "foundationdb.org/none"
+
+	/*
+		Config map constants
+	*/
+
+	// ClusterFileKey defines the key name in the ConfigMap whose value is the cluster file aka connection string
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_constants.go:40]:
```suggestion
	// ClusterFileKey defines the key name in the ConfigMap whose value is the content of the cluster file, the connection string.
```

File: api/v1beta2/foundationdb_env_variables.go [hunk 3]
```diff
@@ -21,6 +21,75 @@
 package v1beta2
 
 const (
-	// EnvNamePublicIP defines the FDB_PUBLIC_IP environment variable name.
+	// EnvNameDNSName specifies the DNS locality (identifies the pod when using DNS in lieu of IP)
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:24]:
```suggestion
	// EnvNameDNSName specifies the DNS locality (identifies the pod when using DNS)
```

File: api/v1beta2/foundationdb_env_variables.go [hunk 4]
```diff
@@ -21,6 +21,75 @@
...
+	EnvNameDNSName = "FDB_DNS_NAME"
+
+	// EnvNameMachineID specifies the Machine ID locality. Also defines fault domain along with EnvNameZoneID
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:27]:
```suggestion
	// EnvNameMachineID specifies the Machine ID locality.
```

The latter is not true, in most cases a single fault domain contains more than one machine.

File: api/v1beta2/foundationdb_env_variables.go [hunk 5]
```diff
@@ -21,6 +21,75 @@
...
+	EnvNameMachineID = "FDB_MACHINE_ID"
+
+	// EnvNameZoneID specifies Zone ID locality. Also defines fault domain along with EnvNameMachineID.
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:30]:
```suggestion
	// EnvNameZoneID specifies Zone ID locality.
```

There can be configurations where a machine is used as fault domain, in most cases a user will define a fault domain that contains more than a single machine (rack, AZ, data hall, etc.).

File: api/v1beta2/foundationdb_env_variables.go [hunk 6]
```diff
@@ -21,6 +21,75 @@
...
+	// The current default value of EnvNameZoneID is the hostname
+	EnvNameZoneID = "FDB_ZONE_ID"
+
+	// EnvNameClusterFile specifies the path to the cluster file / file containing the connection string.
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:34]:
```suggestion
	// EnvNameClusterFile specifies the path to the cluster file.
```

File: api/v1beta2/foundationdb_env_variables.go [hunk 7]
```diff
@@ -21,6 +21,75 @@
...
+	EnvNameTLSKeyFile = "FDB_TLS_KEY_FILE"
+
+	// EnvNameTLSVerifyPeers specifies the peer verification rules for incoming TLS connections to the split-image sidecar.
+	// See https://apple.github.io/foundationdb/tls.html#peer-verification for the format
+	EnvNameTLSVerifyPeers = "FDB_TLS_VERIFY_PEERS"
+
+	// EnvNameFDBNetworkSunsetThing specifies whether to ignore the failure to initialize some of the external clients
+	// TODO FDB 7.3 adds a check for loading external client library, which doesn't work with 6.3.
+	//  Consider remove this option once 6.3 is no longer being used.
+	EnvNameFDBNetworkSunsetThing = "FDB_NETWORK_OPTION_IGNORE_EXTERNAL_CLIENT_FAILURES"
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:61]: This is only used in the testing fixtures right? So I would prefer not to expose this over the `fdbv1beta2` API.

File: api/v1beta2/foundationdb_env_variables.go [hunk 8]
```diff
@@ -21,6 +21,75 @@
...
+
+	// EnvNameClientThreadsPerVersion specifies the number of client threads to be spawned.  Each cluster will be
+	// serviced by a single client thread. Spawns multiple worker threads for each version of the client that is loaded.
+	// Setting this to a number greater than one implies disable_local_client.
+	EnvNameClientThreadsPerVersion = "FDB_NETWORK_OPTION_CLIENT_THREADS_PER_VERSION"
+
+	// EnvNamePublicIP defines the public IP for the split-image-sidecar or unified FDB kubernetes monitor
 	EnvNamePublicIP = "FDB_PUBLIC_IP"
+
+	// EnvNamePodIP specifies the listen address for the split-image-sidecar or unified FDB kubernetes monitor
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:81]: That comment is misleading, nor the kubernetes monitor, nor the sidecar will listen on this address. The value of this env variable will be used to set the listen address of the started fdbserver process.

File: api/v1beta2/foundationdb_env_variables.go [hunk 9]
```diff
@@ -21,6 +21,75 @@
...
+	// EnvNameFDBExternalClientDir specifies path to search for dynamic libraries and adds them to the list of client
+	// libraries for use by the multi-version client API. Must be set before setting up the network.
+	EnvNameFDBExternalClientDir = "FDB_NETWORK_OPTION_EXTERNAL_CLIENT_DIRECTORY"
+
+	// EnvNameClientThreadsPerVersion specifies the number of client threads to be spawned.  Each cluster will be
+	// serviced by a single client thread. Spawns multiple worker threads for each version of the client that is loaded.
+	// Setting this to a number greater than one implies disable_local_client.
+	EnvNameClientThreadsPerVersion = "FDB_NETWORK_OPTION_CLIENT_THREADS_PER_VERSION"
+
+	// EnvNamePublicIP defines the public IP for the split-image-sidecar or unified FDB kubernetes monitor
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:78]: That comment is misleading, nor the kubernetes monitor, nor the sidecar will use this public address. The value of this env variable will be used to set the public address of the started fdbserver process.

File: api/v1beta2/foundationdb_env_variables.go [hunk 10]
```diff
@@ -21,6 +21,75 @@
...
+	// EnvNamePodIP specifies the listen address for the split-image-sidecar or unified FDB kubernetes monitor
+	EnvNamePodIP = "FDB_POD_IP"
+
+	// EnvNamePodName tells the split-image-sidecar or unified FDB kubernetes monitor the name of its pod
+	EnvNamePodName = "FDB_POD_NAME"
+
+	// EnvNamePodNamespace tells the split-image-sidecar or unified FDB kubernetes monitor the K8s namespace it is running in
+	EnvNamePodNamespace = "FDB_POD_NAMESPACE"
+
+	// EnvNameNodeName tells the split-image-sidecar or unified FDB kubernetes monitor the K8s node it is running on
```
Comment (johscheuer) [inline api/v1beta2/foundationdb_env_variables.go:90]:
```suggestion
	// EnvNameNodeName tells the split-image-sidecar or unified FDB kubernetes monitor the K8s node it is running on
```

The split image shouldn't be using that, only the kubernetes monitor uses it.

File: e2e/fixtures/fdb_cluster_specs.go [hunk 11]
```diff
@@ -193,23 +193,23 @@ func (factory *Factory) createPodTemplate(
 					},
 					Env: []corev1.EnvVar{
 						{
-							Name:  "FDB_TLS_CERTIFICATE_FILE",
+							Name:  fdbv1beta2.EnvNameTLSCert,
```
Comment (johscheuer) [inline e2e/fixtures/fdb_cluster_specs.go:196]: I would prefer to define those as constants in the e2e test suite (assuming we only use them in the e2e test suite).

**Review by johscheuer — COMMENTED** [review]
A few comments, otherwise 👍

File: api/v1beta1/foundationdbcluster_types_test.go (hunk 1 above)
Comment (nicmorales9) [reply api/v1beta1/foundationdbcluster_types_test.go:2416]: the dollar sign

File: e2e/fixtures/fdb_cluster_specs.go (hunk 11 above)
Comment (nicmorales9) [reply e2e/fixtures/fdb_cluster_specs.go:196]: that one is used in `NewFdbPodClient` as well, but I will move any which are only used in e2e tests

File: api/v1beta2/foundationdb_env_variables.go (hunk 7 above)
Comment (nicmorales9) [reply api/v1beta2/foundationdb_env_variables.go:61]: yup, gonna move that to e2e

File: api/v1beta1/foundationdbcluster_types_test.go (hunk 1 above)
Comment (johscheuer) [reply api/v1beta1/foundationdbcluster_types_test.go:2416]: Seems like my brain ignored the `$` :)

**Review by johscheuer — APPROVED** [review]
LGTM 👍 Thanks!

File: api/v1beta1/foundationdbcluster_types_test.go (hunk 1 above)
Comment (nicmorales9) [reply api/v1beta1/foundationdbcluster_types_test.go:2416]: oh mine did too until the tests all failed 🤣

**Comment by nicmorales9** [comment]
failures unrelated

**Pull request merged**